src/inmanta/agent/handler.py

```diff
@@ -921,8 +921,8 @@ def execute(self, ctx: HandlerContext, resource: resources.Resource, dry_run: Op
                         self.delete_resource(ctx, desired)
 
                 elif not desired.purged and len(changes) > 0:
-                    ctx.debug("Calling update_resource", changes=changes)
-                    self.update_resource(ctx, dict(changes), desired)
+                    ctx.debug("Calling update_resource", changes=ctx.changes)
+                    self.update_resource(ctx, dict(ctx.changes), desired)
```

Comment:
Why was this changed?

Reply:
Don't we want to pass the sanitized changes to these methods ?

Reply:
mhh, this is a delicate issue:

1. we have two mechanisms to pass on changes (via the context and via the return value). 
2. if I look at all the surrounding code, we only act on those that come from the return value
3. the server however uses the once from the context to trigger reloads and other events 
4. we correctly log what we act on 
5. the corresponding server log may contain many other things.

While the current behavior is not optimal, changing the input to update_resource is a MAJOR change to the behavior of the agent.